pandas/tests/types/test_cast.py

```diff
@@ -218,18 +218,37 @@ def test_numpy_dtypes(self):
             ((np.complex128, np.int32), np.complex128),
             ((np.object, np.float32), np.object),
             ((np.object, np.int16), np.object),
+
+            ((np.dtype('datetime64[ns]'), np.dtype('datetime64[ns]')),
+             np.dtype('datetime64[ns]')),
+            ((np.dtype('timedelta64[ns]'), np.dtype('timedelta64[ns]')),
+             np.dtype('timedelta64[ns]')),
+
+            ((np.dtype('datetime64[ns]'), np.dtype('datetime64[ms]')),
+             np.dtype('datetime64[ns]')),
+            ((np.dtype('timedelta64[ms]'), np.dtype('timedelta64[ns]')),
+             np.dtype('timedelta64[ns]')),
+
+            ((np.dtype('datetime64[ns]'), np.dtype('timedelta64[ns]')),
+             np.object),
+            ((np.dtype('datetime64[ns]'), np.int64), np.object)
         )
         for src, common in testcases:
             self.assertEqual(_find_common_type(src), common)
 
     def test_pandas_dtypes(self):
-        # TODO: not implemented yet
-        with self.assertRaises(TypeError):
-            self.assertEqual(_find_common_type([CategoricalDtype()]),
-                             CategoricalDtype)
-        with self.assertRaises(TypeError):
-            self.assertEqual(_find_common_type([DatetimeTZDtype()]),
-                             DatetimeTZDtype)
+        dtype = CategoricalDtype()
+        self.assertEqual(_find_common_type([dtype]), 'category')
+        self.assertEqual(_find_common_type([dtype, dtype]), 'category')
+        self.assertEqual(_find_common_type([np.object, dtype]), np.object)
+
+        dtype = DatetimeTZDtype(unit='ns', tz='US/Eastern')
```

Comment:
add a test for a datetime64[ns] w/tz against datetime64[ns]. (and maybe with np.object)